### ISSUE TITLE
Update script does not restart php7.x-fpm on PHP module update

### ISSUE THREAD

**Issue by hatclub** [nextcloud/vm#715]
## Steps to reproduce
1. run /var/scripts/update.sh with a PECL redis update pending

## Expected behaviour
Update script should restart php7.x-fpm if the PHP redis module is recompiled so that things continue working

## Actual behaviour
Update script does not restart php7.x-fpm, breaking server - webserver starts returning error 500.

## Server configuration

**Server version:** (see your admin page)
14.0.3

**How did you install the VM?** (Scripted install from master **OR** Released version)
Was installed using released VM template 4 days ago, first update today

## Network
n/a

#### Log file (/mnt/ncdata/nextcloud.log)
{"reqId":"W-YW4uVjrvBctxYlSqsQVQAAAAc","level":3,"time":"2018-11-22T02:39:30+00:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"--","app":"core","method":"GET","url":"\/favicon.ico","message":{"Exception":"OC\\HintException","Message":"Memcache \\OC\\Memcache\\Redis not available for local cache","Code":0,"Trace":[{"file":"\/var\/www\/nextcloud\/lib\/private\/Server.php","line":502,"function":"__construct","class":"OC\\Memcache\\Factory","type":"->","args":["53a6449a79707a4b02869234c1273086",{"__class__":"OC\\Log"},"\\OC\\Memcache\\Redis","\\OC\\Memcache\\Redis","\\OC\\Memcache\\Redis"]},{"file":"\/var\/www\/nextcloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php","line":118,"function":"OC\\{closure}","class":"OC\\Server","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":117,"function":"offsetGet","class":"Pimple\\Container","type":"->","args":["OC\\Memcache\\Factory"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/ServerContainer.php","line":132,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Memcache\\Factory"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/Server.php","line":974,"function":"query","class":"OC\\ServerContainer","type":"->","args":["OC\\Memcache\\Factory"]},{"file":"\/var\/www\/nextcloud\/3rdparty\/pimple\/pimple\/src\/Pimple\/Container.php","line":118,"function":"OC\\{closure}","class":"OC\\Server","type":"->","args":["*** sensitive parameters replaced ***"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":117,"function":"offsetGet","class":"Pimple\\Container","type":"->","args":["OC\\Template\\JSCombiner"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/ServerContainer.php","line":132,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Template\\JSCombiner"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/TemplateLayout.php","line":319,"function":"query","class":"OC\\ServerContainer","type":"->","args":["OC\\Template\\JSCombiner"]},{"file":"\/var\/www\/nextcloud\/lib\/private\/TemplateLayout.php","line":160,"function":"findJavascriptFiles","class":"OC\\TemplateLayout","type":"::","args":[["vendor\/core","js\/merged-template-prepend","search\/js\/search","core\/l10n\/en_GB","search\/l10n\/en_GB","search\/js\/searchprovider","js\/jquery-ui-fixes","js\/files\/fileinfo","js\/files\/client","js\/contactsmenu"]]},{"file":"\/var\/www\/nextcloud\/lib\/private\/legacy\/template.php","line":210,"function":"__construct","class":"OC\\TemplateLayout","type":"->","args":["error",""]},{"file":"\/var\/www\/nextcloud\/lib\/private\/Template\/Base.php","line":132,"function":"fetchPage","class":"OC_Template","type":"->","args":[]},{"file":"\/var\/www\/nextcloud\/lib\/private\/legacy\/template.php","line":359,"function":"printPage","class":"OC\\Template\\Base","type":"->","args":[]},{"file":"\/var\/www\/nextcloud\/index.php","line":61,"function":"printExceptionErrorPage","class":"OC_Template","type":"::","args":[{"__class__":"OC\\HintException"},500]}],"File":"\/var\/www\/nextcloud\/lib\/private\/Memcache\/Factory.php","Line":99,"Hint":"Is the matching PHP module installed and enabled?","CustomMessage":"--"},"userAgent":"Mozilla\/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit\/537.36 (KHTML, like Gecko) Chrome\/70.0.3538.110 Safari\/537.36","version":"14.0.3.0"}

#### Installation log (command output)
n/a

#### Screenshoots
n/a

**Comment by hatclub** [comment]
Should the update script pull down the latest build of itself, or is that a separate (manual) undertaking?

**Comment by enoch85** [comment]
So this had nothing to do with the redis update, rather Modesec?

Modsec is what Modsec is, and I don't recommend enabling it if you're not familiar with it since it can cause "damage" to your system.

I've seen multiple reports by users where it doesn't allow the sync client to sync all files for example. Hence the clear warning when installing it.

**Comment by hatclub** [comment]
No, the original issue with the redis updated was caused prior to the disk being full (I hadn't uploaded anything for modsec to log).

I was only mentioning that I ran into problems testing the latest revision of the updater which happened to be modsec log related (as it logged the payload of *every file uploaded*). I am familiar with modsec, and it's not actually in blocking mode. My question was only if there is an 'official' way in your appliance to uninstall this, or is just removing the package enough - i.e. if I just remove the packages with apt, is update.sh going to try to reinstall it because it thinks it should be installed, or does your code not maintain it's own state of installation options?

**Comment by enoch85** [comment]
If you don't want Modsec then just uninstall it. 

The VM is based on a regular Ubuntu Server 18.04 installation, hence uninstalling with apt will not add it back again. Think of it as your own regular server with some extras for easier management.

The update script in general servers the user with updates for stuff outside of the apt scope if it's installed, but it checks before if it's installed. Docker, Redis and Adminer are such cases e.g.
